src/test/munit/directory-listener-functional-test-case.xml

```diff
@@ -0,0 +1,396 @@
+            <file:write config-ref="file-list-config" path="withMatcher/matchme.txt">
+                <file:content>Dr. Manhattan</file:content>
+            </file:write>
+            <file:write config-ref="file-list-config" path="withMatcher/watchme.txt">
+                <file:content>who watches the watchmen?w></file:content>
+            </file:write>
+        </munit:behavior>
+        <munit:execution>
+            <try>
+                <until-successful maxRetries="10" millisBetweenRetries="1000" >
+                    <munit-tools:retrieve key="messages"/>
+                    <validation:validate-size value="#[payload]" min="1" max="1"/>
+                </until-successful>
+                <error-handler >
+                    <on-error-continue enableNotifications="false" logException="false" type="ANY">
+                        <munit-tools:retrieve key="messages"/>
+                    </on-error-continue>
+                </error-handler>
+            </try>
+            <set-variable value="#[payload]" variableName="files"/>
+            <try>
+                <until-successful maxRetries="10" millisBetweenRetries="1000">
+                    <munit-tools:retrieve key="messages" />
+                    <validation:validate-size value="#[payload]" min="2"/>
+                </until-successful>
+                <munit-tools:fail/>
+                <error-handler>
+                    <on-error-continue enableNotifications="false" logException="false" type="MULE:RETRY_EXHAUSTED"/>
+                </error-handler>
+            </try>
+        </munit:execution>
+        <munit:validation>
+            <munit-tools:assert-that expression="#[vars.files[0].attributes.fileName]" is="#[MunitTools::equalTo('matchme.txt')]"/>
+        </munit:validation>
+    </munit:test>
+
+    <flow name="source-non-recursive-test">
+        <file:listener config-ref="file-list-config" directory="matcherless" recursive="false" autoDelete="true">
+            <scheduling-strategy>
+                <fixed-frequency/>
+            </scheduling-strategy>
+        </file:listener>
+        <flow-ref name="storeMessages"/>
+    </flow>
+    <munit:test name="non-recursive-test" description="Verifies that files created in subdirs are not picked">
+        <munit:enable-flow-sources >
+            <munit:enable-flow-source value="source-non-recursive-test"/>
+        </munit:enable-flow-sources>
+        <munit:behavior>
+            <file:write config-ref="file-list-config" path="matcherless/subdir/watchme.txt">
+                <file:content >who watches the watchmen?</file:content>
+            </file:write>
+            <file:write config-ref="file-list-config" path="matcherless/nonRecursive.txt">
+                <file:content >you shall not recurse</file:content>
+            </file:write>
+        </munit:behavior>
+        <munit:execution >
+            <try>
+                <until-successful maxRetries="10" millisBetweenRetries="1000" >
```

Comment:
The behavior of getting sure that 1 file is picked, and later check that no more than 1 file was picked is used in another test also. This can be extracted to a flow and call it with a flow ref to avoid repetitions